pennylane/optimize/shot_adaptive.py

```diff
@@ -29,10 +30,25 @@ class ShotAdaptiveOptimizer(GradientDescentOptimizer):
     of the parameter-shift gradient, this optimizer frugally distributes a shot
     budget across the partial derivatives of each parameter.
 
-    In addition, if computing the expectation value of a Hamiltonian using
-    :class:`~.ExpvalCost`, weighted random sampling can be used to further
-    distribute the shot budget across the local terms from which the Hamiltonian
-    is constructed.
+    In addition, weighted random sampling can be used to further distribute the
+    shot budget across the local terms from which the Hamiltonian is constructed.
+
+    .. warning::
+
+        ``ExpvalCost`` is deprecated. Instead, it is recommended to simply
+        pass Hamiltonians to the :func:`~pennylane.expval` function inside QNodes.
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            def ansatz(params):
+                some_qfunc(params)
+                return qml.expval(Hamiltonian)
+
+        In order to optimize the Hamiltonian evaluation taking into account commuting terms, use the ``grouping_type`` keyword in :class:`~.Hamiltonian`.

```

Comment:
Does the grouping effect the shot adaptive optimizer at all?

Reply:
well now that you mention it... it kinda breaks it actually.
- on master with ExpvalCost, it only uses the shots associated with the first term/coeff, scales by that coeff, then raises some mysterious warning
- on this branch with ExpvalCost, autograd fails outright when trying to reshape the resulting jacobian
- with QNode (both versions), the grouping has no effect

this seems like a whole mess - but I'll remove the suggestion to set `grouping_type` for this PR